bai2/constants.py

```diff
@@ -133,6 +133,7 @@ class TypeCodeLevel(Enum):
     TypeCode('150', TypeCodeTransaction.credit, TypeCodeLevel.summary, 'Total Preauthorized Payment Credits'),
     TypeCode('155', TypeCodeTransaction.credit, TypeCodeLevel.detail, 'Preauthorized Draft Credit'),
     TypeCode('156', TypeCodeTransaction.credit, TypeCodeLevel.detail, 'Item in PAC Deposit'),
+    TypeCode('158', TypeCodeTransaction.credit, TypeCodeLevel.detail, 'Real Time Payment Credit'),
```

Comment:
Do you know if these two new codes are used by other organisations? I ask in case they have different meanings elsewhere and that, for your use case, we might be better off making it easier for `bai2` _clients_ to extend these enumerations.

Reply:
I know that they are used by the Silicon Valley bank since we have received files from them with the 158 detail code, that causes the parsing with the existing package to fail.  I don't know if they are used by other organizations.

We have not seen the 458 code, but it seemed related which is why I added it.

I am not sure what you are suggesting to make it easier for bai2 clients to extend these enumerations.  I don't might doing the implementation if you have a design pattern that you would like me to follow.

Reply:
Here are three different URLs that all describe the code 158 as Real-Time Payment credits

http://connect-content.us.hsbc.com/hsbc_pcm/onetime/19_us_rtp_stmt_details.html#:~:text=Real%2DTime%20Payment%20credits%20will%20report%20with%20a%20transaction,158%20on%20your%20BAI%20statement.&text=TIME%2F08%3A09-,What%20this%20means%20for%20you%3A,payment%20mechanism%20for%20the%20US.
https://x9.org/wp-content/uploads/2018/07/BTRS-BAI2-Transaction-Codes-and-Format-to-Support-Real-Time-Payments.pdf
https://x9.org/standards/btrs/

![image](https://user-images.githubusercontent.com/119447023/219693080-efad61cf-6ccf-4f97-98c6-424c2a85edde.png)